### PR TITLE
update ndarray stack Doc for #11925

### PR DESCRIPTION
**Pull request by liyujiel**
## Description ##
Update Doc for ndarray stack. 
https://github.com/apache/incubator-mxnet/issues/11925

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [x] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [x] API doc for ndarray.stack

### CODE REVIEW

**Comment by nswamy** [comment]
Thanks for your contribution, The documentation in C code is intended to be used by all language bindings(C++, Scala, Java, Julia, Python, etc.,) 
To write a specific Python doc with example, please follow this 
https://github.com/apache/incubator-mxnet/blob/master/python/mxnet/ndarray_doc.py#L68

**Comment by liyujiel** [comment]
Get error: *** Error in `python3.6': double free or corruption (fasttop): 0x00007f58cc000950 ***

I think there is a problem with CI. Same problem: https://github.com/tensorflow/tensorflow/issues/6968

File: src/operator/tensor/matrix_op.cc [hunk 1]
```diff
@@ -829,8 +829,8 @@ will be the last dimension.
 
 Examples::
 
-  x = [1, 2]
-  y = [3, 4]
+  x = mx.nd.array([1, 2])
```
Comment (nswamy) [inline src/operator/tensor/matrix_op.cc]: please remove changes to these two lines, since mx.nd is only for Python and your example covers this.

**Comment by liyujiel** [comment]
@szha 

Get error: *** Error in `python3.6': double free or corruption (fasttop): 0x00007f58cc000950 ***

I think there is a problem with CI. Same problem: tensorflow/tensorflow#6968

**Comment by szha** [comment]
@liyujiel can you do a rebase on the current master?

**Pull request merged**